src/Raven.Server/Documents/ETL/Providers/Raven/RavenEtlDocumentTransformer.cs

```diff
@@ -356,6 +310,97 @@ bool ShouldFilterOutDeletion()
             _commands.AddRange(_currentRun.GetCommands());
         }
 
+
+
+        private void AddCounters(LazyStringValue docId, BlittableJsonReaderObject counterGroupDocument, string function = null)
+        {
+            if (!counterGroupDocument.TryGet(CountersStorage.Values, out BlittableJsonReaderObject counters))
+                return;
+
+            var prop = new BlittableJsonReaderObject.PropertyDetails();
+            for (var i = 0; i < counters.Count; i++)
+            {
+                counters.GetPropertyByIndex(i, ref prop);
+
+                if (GetCounterValueAndCheckIfShouldSkip(docId, function, prop, out long value, out bool delete))
+                    continue;
+
+                if (delete)
+                    _currentRun.DeleteCounter(docId, prop.Name);
+                else
+                    _currentRun.AddCounter(docId, prop.Name, value);
+            }
+        }
+
+        private List<CounterOperation> GetCounterOperationsFor(RavenEtlItem item)
+        {
+            var cgd = Database.DocumentsStorage.CountersStorage.GetCounterValuesForDocument(Context, item.DocumentId);
+
+            if (!cgd.Values.TryGet(CountersStorage.Values, out BlittableJsonReaderObject counters))
+                return null;
+            var counterOperations = new List<CounterOperation>();
+            var prop = new BlittableJsonReaderObject.PropertyDetails();
+            for (var i = 0; i < counters.Count; i++)
+            {
+                counters.GetPropertyByIndex(i, ref prop);
+
+                if (GetCounterValueAndCheckIfShouldSkip(item.DocumentId, null, prop, out long value, out bool delete))
+                    continue;
+
+                if (delete)
+                {
+                    counterOperations.Add(new CounterOperation
+                    {
+                        Type = CounterOperationType.Delete,
+                        CounterName = prop.Name,
+                    });
+                }
+                else
+                {
+                    counterOperations.Add(new CounterOperation
+                    {
+                        Type = CounterOperationType.Put,
+                        CounterName = prop.Name,
+                        Delta = value
+                    });
+                }
+            }
+
+            return counterOperations;
+        }
+
+        private bool GetCounterValueAndCheckIfShouldSkip(LazyStringValue docId, string function, BlittableJsonReaderObject.PropertyDetails prop, out long value, out bool delete)
+        {
+            value = 0;
+
+            if (prop.Value is LazyStringValue)
```

Comment:
Does value being `LazyStringValue` mean that counter was deleted?

Reply:
yes

Reply:
That's tricky. Could you put a comment so one won't be surprised as I was?